### PR TITLE
Update to 3.2.1

### PR DESCRIPTION
**Pull request by nicoddemus**

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -25,7 +25,7 @@ source:
     - PR_1971.diff
```
Comment (jakirkham) [inline recipe/meta.yaml]: Try dropping the patches and deleting the associated files. They appear to already be included in 3.2.1.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -8,7 +8,7 @@ package:
 source:
   fn: PyInstaller-{{ version }}.tar.gz
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
```
Comment (jakirkham) [inline recipe/meta.yaml]: Looks like they uploaded a `.tar.bz2` instead of a `.tar.gz`. Changing this extension to `.tar.bz2` should fix it.

Comment (nicoddemus) [inline recipe/meta.yaml]: Ahh thanks, did not notice!

File: recipe/meta.yaml (hunk 1 above)
Comment (nicoddemus) [inline recipe/meta.yaml]: Thanks!

**Comment by nicoddemus** [comment]
Windows broke because of the `pyi-grab_version` script:

```
Traceback (most recent call last):
  File "C:\bld\pyinstaller_1488926523578\_t_env\Scripts\pyi-grab_version-script.py", line 5, in <module>
    sys.exit(PyInstaller.utils.cliutils.grab_version.run())
  File "C:\bld\pyinstaller_1488926523578\_t_env\lib\site-packages\PyInstaller\utils\cliutils\grab_version.py", line 19, in run
    PyInstaller.log.init()
UnboundLocalError: local variable 'PyInstaller' referenced before assignment
```

This is the `grab_version.py` script inside the package:

```python
#-----------------------------------------------------------------------------
# Copyright (c) 2013-2016, PyInstaller Development Team.
#
# Distributed under the terms of the GNU General Public License with exception
# for distributing bootloader.
#
# The full license is in the file COPYING.txt, distributed with this software.
#-----------------------------------------------------------------------------


import codecs
import os
import argparse

import PyInstaller.log


def run():
    PyInstaller.log.init()

    parser = argparse.ArgumentParser(
        epilog = ('The printed output may be saved to a file, edited and '
                  'used as the input for a version resource on any of the '
                  'executable targets in an Installer spec file.'))
    parser.add_argument('exe_file', metavar='exe-file',
                        help="full pathname of a Windows executable")
    parser.add_argument('out_filename', metavar='out-filename', nargs='?',
                        default='file_version_info.txt',
                        help=("filename where the grabbed version info "
                              "will be saved"))

    args = parser.parse_args()

    try:
        import PyInstaller.utils.win32.versioninfo
        vs = PyInstaller.utils.win32.versioninfo.decode(args.exe_file)
        fp = codecs.open(args.out_filename, 'w', 'utf-8')
        fp.write(unicode(vs))
        fp.close()
        print(('Version info written to: %s' % args.out_filename))
    except KeyboardInterrupt:
        raise SystemExit("Aborted by user request.")

if __name__ == '__main__':
    run()
```

[Current `master`](https://github.com/pyinstaller/pyinstaller/blob/develop/PyInstaller/utils/cliutils/grab_version.py) looks fixed:

```python
#-----------------------------------------------------------------------------
# Copyright (c) 2013-2017, PyInstaller Development Team.
#
# Distributed under the terms of the GNU General Public License with exception
# for distributing bootloader.
#
# The full license is in the file COPYING.txt, distributed with this software.
#-----------------------------------------------------------------------------


import codecs
import os
import argparse

import PyInstaller.log


def run():
    parser = argparse.ArgumentParser(
        epilog = ('The printed output may be saved to a file, edited and '
                  'used as the input for a version resource on any of the '
                  'executable targets in an Installer spec file.'))
    parser.add_argument('exe_file', metavar='exe-file',
                        help="full pathname of a Windows executable")
    parser.add_argument('out_filename', metavar='out-filename', nargs='?',
                        default='file_version_info.txt',
                        help=("filename where the grabbed version info "
                              "will be saved"))

    args = parser.parse_args()

    try:
        import PyInstaller.utils.win32.versioninfo
        vs = PyInstaller.utils.win32.versioninfo.decode(args.exe_file)
        fp = codecs.open(args.out_filename, 'w', 'utf-8')
        fp.write(unicode(vs))
        fp.close()
        print(('Version info written to: %s' % args.out_filename))
    except KeyboardInterrupt:
        raise SystemExit("Aborted by user request.")

if __name__ == '__main__':
    run()
```

This was fixed in [10f5e44](https://github.com/pyinstaller/pyinstaller/commit/10f5e44604e163383f8f32ab05307edd69ecab1c), although for a different reason it seems.

`PyInstaller-3.2.1` installed from PyPI has the same problem:

```
{installer-env} X:\temp>pyi-grab_version -h
Traceback (most recent call last):
  File "d:\programming\python35\Lib\runpy.py", line 170, in _run_module_as_main
    "__main__", mod_spec)
  File "d:\programming\python35\Lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "X:\temp\installer-env\Scripts\pyi-grab_version.exe\__main__.py", line 9, in <module>
  File "x:\temp\installer-env\lib\site-packages\PyInstaller\utils\cliutils\grab_version.py", line 19, in run
    PyInstaller.log.init()
UnboundLocalError: local variable 'PyInstaller' referenced before assignment
```

Since this is not a problem with the conda package and the PyPI package has the same issue, I think we should comment that call on the recipe for `3.2.1` to create the conda package, but make sure to uncomment it for `3.2.2`. What do you think @jakirkham?

**Comment by nicoddemus** [comment]
`set_version.py` on the package has the same problem, and has also been fixed in `master`.

**Comment by jakirkham** [comment]
Would it make sense to just patch them? Given the fixes are known, this should be relatively straightforward. That way users expecting these to work won't have any surprises.

**Comment by nicoddemus** [comment]
Not sure, because those fixes are not available in PyPI's `3.2.1`, but it's your call. 😁

**Comment by jakirkham** [comment]
As we aren't adding new functionality, but fixing existing functionality, patching should be ok. Especially for something like bad log initialization.

**Comment by nicoddemus** [comment]
No problem! Decided to add a patch for the entire commit because it seems to fix the same problem in other tools as well.

**Comment by nicoddemus** [comment]
Does it look good to you @jakirkham? Can we merge once Travis passes?

**Comment by nicoddemus** [comment]
Travis still hasn't even started... I know Travis (OSX) takes longer than the others, but is this normal? 😟

**Comment by jakirkham** [comment]
Sadly it has been taking a day or two. Appears Travis CI added some more workers recently. Though I think the big problem is we share the queue with all orgs unlike other CIs where each org gets a queue. In any event, it appears to have completed so we are good to go.

**Comment by jakirkham** [comment]
Thanks @nicoddemus.

**Comment by nicoddemus** [comment]
:+1:

**Pull request merged**